packages/itwinui-react/src/core/Popover/Popover.tsx

```diff
@@ -294,36 +307,68 @@ export const Popover = React.forwardRef((props, forwardedRef) => {
       }))}
 
       {popover.open ? (
-        <Portal portal={portal}>
-          <FloatingPortal>
-            <ThemeProvider
-              portalContainer={popoverElement} // portal nested popovers into this one
+        <PopoverPortal portal={portal}>
+          <ThemeProvider
+            portalContainer={popoverElement} // portal nested popovers into this one
+          >
+            <DisplayContents />
+            <FloatingFocusManager
+              context={popover.context}
+              modal={false}
+              initialFocus={popover.refs.floating}
             >
-              <FloatingFocusManager
-                context={popover.context}
-                modal={false}
-                initialFocus={popover.refs.floating}
+              <Box
+                className={cx(
+                  { 'iui-popover-surface': applyBackground },
+                  className,
+                )}
+                aria-labelledby={
+                  !hasAriaLabel
+                    ? popover.refs.domReference.current?.id
+                    : undefined
+                }
+                {...popover.getFloatingProps(rest)}
+                ref={popoverRef}
               >
-                <Box
-                  className={cx(
-                    { 'iui-popover-surface': applyBackground },
-                    className,
-                  )}
-                  aria-labelledby={
-                    !hasAriaLabel
-                      ? popover.refs.domReference.current?.id
-                      : undefined
-                  }
-                  {...popover.getFloatingProps(rest)}
-                  ref={popoverRef}
-                >
-                  {content}
-                </Box>
-              </FloatingFocusManager>
-            </ThemeProvider>
-          </FloatingPortal>
-        </Portal>
+                {content}
+              </Box>
+            </FloatingFocusManager>
+          </ThemeProvider>
+        </PopoverPortal>
       ) : null}
     </>
   );
 }) as PolymorphicForwardRefComponent<'div', PopoverPublicProps>;
+
+// ----------------------------------------------------------------------------
+
+const PopoverPortal = ({
+  children,
+  portal = true,
+}: React.PropsWithChildren<PortalProps>) => {
+  const portalTo = usePortalTo(portal);
+
+  return (
+    <FloatingPortal root={portalTo}>
+      <DisplayContents />
+      {children}
+    </FloatingPortal>
+  );
+};
+
+// ----------------------------------------------------------------------------
+
+/** Applies `display: contents` to the parent div. */
+const DisplayContents = React.memo(() => {
+  return (
+    <ShadowRoot
+      css={`
+        :host {
+          display: contents;
```

Comment:
[`display: contents`](https://developer.mozilla.org/en-US/docs/Web/CSS/display#box) makes it so that the element becomes "see-through" for the purposes of layout and sizing.

So it shouldn't affect 99% of the cases, but I'm thinking of this one very specific case that I want to continue to support. If the user uses `flex` or `grid` layout on the ThemeProvider, the portaled elements shouldn't participate in the flex/grid layout.

```js
<ThemeProvider style={{ display: "grid" }}>
  <Header>…</Header>
  <SideNavigation>…</SideNavigation>
  <main>…</main>
  <Footer>…</Footer>
</ThemeProvider>
```